app/modules/mqtt.c

```diff
@@ -971,22 +982,21 @@ static int mqtt_socket_connect( lua_State* L )
     return luaL_error(L, "already connected");
   }
 
-  if(mud->pesp_conn){   //TODO: should I free tcp struct directly or ask user to call close()???
-    mud->pesp_conn->reverse = NULL;
-    if(mud->pesp_conn->proto.tcp)
-      c_free(mud->pesp_conn->proto.tcp);
-    mud->pesp_conn->proto.tcp = NULL;
-    c_free(mud->pesp_conn);
-    mud->pesp_conn = NULL;
+  struct espconn *pesp_conn = mud->pesp_conn;
+  if(!pesp_conn) {
+    pesp_conn = mud->pesp_conn = (struct espconn *)c_zalloc(sizeof(struct espconn));
+  } else {
+    espconn_delete(pesp_conn);
```

Comment:
Is the state of `pesp_conn` equivalent after the two `if` branches? Can it be re-used below if it's deleted here? The API guide is not clear in this respect IMO.